tests/ably/setup/sandbox.ts

```diff
@@ -0,0 +1,93 @@
+import path from "path";
+import fs from 'fs';
+import { httpReqFunction } from "../../../src/channel/ably/utils";
+
+let restHost = 'sandbox-rest.ably.io';
+const tlsPort = 443;
+const toBase64 = (text: string) => Buffer.from(text, 'binary').toString('base64');
+
+const loadJsonData = (dataPath, callback) => {
+    fs.readFile(path.join(__dirname, dataPath), (err, data: Buffer) => {
+        if (err) {
+            callback(err);
+            return;
+        }
+        try {
+            data = JSON.parse(data.toString());
+        } catch (e) {
+            callback(e);
+            return;
+        }
+        callback(null, data);
+    });
+}
+
+const httpReq = httpReqFunction();
+
+function prefixDomainWithEnvironment(domain, environment) {
+    if (environment.toLowerCase() === 'production') {
+        return domain;
+    } else {
+        return environment + '-' + domain;
+    }
+}
+
+const creatNewApp = (callback) => {
```

Comment:
This sandbox API implementation is probably overkill since it was written in ably-js to be isomorphic and highly configurable so that it can run against a local realtime instance. I'd consider adding an HTTP client like 'got' from NPM as a devDependency and doing something more along the lines of:

```ts
import * as Ably from 'ably';
import got from 'got';

export class TestApp {
  id: string;
  key: string;

  static async create() {
    const url = 'https://sandbox-rest.ably.io/apps';
    const body = { keys: [ {} ] };

    const res: {
      appId: string,
      keys: { keyStr: string}[],
    } = await got.post(url, { json: body }).json();

    return new TestApp(res.appId, res.keys[0].keyStr);
  }

  constructor(id: string, key: string) {
    this.id = id;
    this.key = key;
  }

  client(clientId: string): Ably.Realtime {
    return new Ably.Realtime({
      key:         this.key,
      environment: 'sandbox',
      clientId,
    });
  }

  delete() {
    return got.delete(`https://sandbox-rest.ably.io/apps/${this.id}?key=${this.key}`);
  }
}
```

Reply:
done